tools/ports/ogg.py

```diff
@@ -28,7 +29,7 @@ def create(final):
     shutil.rmtree(dest_path, ignore_errors=True)
     shutil.copytree(source_path, dest_path)
 
-    open(os.path.join(dest_path, 'include', 'ogg', 'config_types.h'), 'w').write(config_types_h)
+    Path(dest_path, 'include', 'ogg', 'config_types.h').write_text(config_types_h)
```

Comment:
You have access to `shared` here and in the other `ports` files.

Reply:
Sorry, I have reached my limit for manual and tedious changes between `Path.read_text` `open().read()` and  `shared.read_file` for this PR :) I understand not using Path would save 1 import here,  but in the end it is a stdlib module, it's readable and has no other undesirable side effects. Please feel free to push the change if that's a blocker.

Reply:
Also with `read_file` I would have to either revert back to using `os.path.join` or change the `read_binary signature to accept path segments, and the latter gives me the feeling that this PR is never going to end :)